### ISSUE TITLE
Standard In

### ISSUE THREAD

**Issue by yonaskolb** [jakeheis/SwiftCLI#47]
Is launching a process via a Task and then having that read from standard in supported? It doesn't seem to work by default.

**Comment by jakeheis** [comment]
This actually seems like it might be a limitation on `Foundation.Process` which `Task` is built on. Will investigate further, may also consider implementing Task directly with `posix_spawn`

**Comment by yonaskolb** [comment]
Other discussion here on SwiftShell for reference https://github.com/kareman/SwiftShell/issues/26

**Comment by yonaskolb** [comment]
This seems to do the trick.
```
let inPipe = Pipe()
inPipe.fileHandleForWriting.writeabilityHandler = { handler in
    if let d = readLine(strippingNewline: false)?.data(using: .utf8) {
        handler.write(d)
    } else {
        handler.closeFile()
    }
}
process.standardInput = inPipe
```

**Comment by jakeheis** [comment]
The problem with this approach is that it continues trying to write lines to the pipe after the process ends. You can see the issue with something like:

```swift
let proc = Process()
proc.launchPath = "/bin/bash"

let inPipe = Pipe()
inPipe.fileHandleForWriting.writeabilityHandler = { handler in
    if let d = readLine(strippingNewline: false)?.data(using: .utf8) {
        handler.write(d)
    } else {
        handler.closeFile()
    }
}

proc.standardInput = inPipe
proc.launch()
proc.waitUntilExit()

let next = readLine()
print("got", next as Any)
```

It will work initially, but once you type "exit" to quit the bash process, the next line you type should get read into the `next` var, but instead this tries to write it to the (now closed) `inPipe` and the process is killed by a SIGPIPE signal.

The ideal fix to this in my mind would be to use `tcsetpgrp` which just transfers control of a tty's stdin to a different process:

```swift
let proc = Process()
proc.launchPath = "/bin/bash"

proc.launch()
tcsetpgrp(STDIN_FILENO, proc.processIdentifier)
proc.waitUntilExit()

tcsetpgrp(STDIN_FILENO, getpid())
let next = readLine()
print("got", next as Any)
```

but `Process` does something internally that makes this only partially work. Input-wise this works perfect, the only issue is that for some reason when `tcsetpgrp` is called to transfer stdin control back to the main process, a SIGSTOP is issued:

```shell
> swift run
bash-3.2$ ls
Basher.xcodeproj Package.swift    README.md        Sources          Tests
bash-3.2$ exit
exit
[2]  + 27229 suspended (tty output)  swift run
> fg
[2]  - 27229 continued  swift run
hello
got Optional("hello")
```

**Comment by jakeheis** [comment]
With a small change I was able to get this working on macOS but it completely breaks down on Linux unfortunately

```swift
let proc = Process()
proc.launchPath = "/bin/bash"
proc.launch()

tcsetpgrp(STDIN_FILENO, proc.processIdentifier)

proc.waitUntilExit()

signal(SIGTTOU, SIG_IGN)
tcsetpgrp(STDIN_FILENO, getpid())
signal(SIGTTOU, SIG_DFL)

let next = readLine()
print("got", next as Any)
```

**Comment by jakeheis** [comment]
Actually it works on Linux without anything fancy! So I believe this solution could work

**Comment by jakeheis** [comment]
Released as `4.2.1`
